src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.h

```diff
@@ -0,0 +1,194 @@
+
+        // TODO: #258 polygon support
+        void poly();
+
+        /**
+         * Request a message to draw an arc. The origin is the center point
+         *
+         * @param layer: The layer name this shape is being drawn to
+         * @param cx: Arc's center point X
+         * @param cy: Arc's center point Y
+         * @param radius: Arc's radius
+         * @param theta_start: The starting angle of the arc in rad (where 0 is pointed to +X horizontal heading, and PI/2 is south)
+		 * @param theta_end: The ending angle of the arc in rad
+         * @param draw_style: the drawing style of the shape
+         * @param draw_transform: the transformation of the shape
+         */
+        void arc(const std::string& layer, double cx, double cy, double radius, double theta_start, double theta_end, DrawStyle draw_style = DrawStyle(), DrawTransform draw_transform = DrawTransform());
+
+        /**
+         * Request a message to draw a line. The origin is the first point
+         *
+         * @param layer: The layer name this shape is being drawn to
+         * @param x1: Starting point X
+         * @param y1: Starting point Y
+         * @param x2: Ending point X
+         * @param y2: Ending point Y
+         * @param draw_style: the drawing style of the shape
+         * @param draw_transform: the transformation of the shape
+         */
+        void line(const std::string& layer, double x1, double y1, double x2, double y2, DrawStyle draw_style = DrawStyle(), DrawTransform draw_transform = DrawTransform());
+
+	private:
+
+		/**
+		 * Constructor; initializes an empty layers map then populates it
+		 */
+		explicit VisualizerMessenger()
+		: m_draw_layers()
+		{
+			buildLayers();
+		}
+
+		/**
+		 * Populates the layers map with some pre-defined layers
+		 */
+		void buildLayers();
+
+		/**
+		 * Helper function for the shape draw methods that copies the attributes in DrawStyle
+		 * struct into the actual shape message
+		 * 
+		 * @param shape_msg: Reference to the shape message
+		 * @param style: The DrawStyle to be extracted
+		 */
+		void applyDrawStyleToMsg(ShapeMsg& shape_msg, DrawStyle& style);
+
+        /**
+         * Helper function for the shape draw methods that copies the attributes in
+         * DrawTransform struct into the actual shape message
+         *
+         * @param shape_msg: Reference to the shape message
+         * @param transform: The DrawTransform struct to be extracted
+         */
+        void applyDrawTransformToMsg(ShapeMsg& shape_msg, DrawTransform& transform);
+
+		/**
+		 * Helper function that checks if the layer exists, 
+		 * and push the shape message to the corresponding layer's shape vector.
+		 * 
+		 * @param layer: The name of the layer to push to
+		 * @param shape: The shape message
+		 */
+		void addShapeToLayer(const std::string& layer, ShapeMsg& shape);
+
+	private:
+
+		// string to LayerMsg map
+		LayerMsgMap m_draw_layers;
```

Comment:
could we rename this to `layer_name_to_layer_map`? I know it's more verbose, but I'm not sure what `m_draw_layers` stands for, and I think it's better that we be explicit?